plugins/inputs/redis/redis.go

```diff
@@ -658,17 +659,24 @@ func gatherErrorstatsLine(
 	line string,
 	acc telegraf.Accumulator,
 	globalTags map[string]string,
+	log telegraf.Logger,
 ) {
 	tags := make(map[string]string, len(globalTags)+1)
 	for k, v := range globalTags {
 		tags[k] = v
 	}
 	tags["err"] = strings.TrimPrefix(name, "errorstat_")
-	kv := strings.Split(line, "=")
-	ival, err := strconv.ParseInt(kv[1], 10, 64)
-	if err == nil {
-		fields := map[string]interface{}{"total": ival}
-		acc.AddFields("redis_errorstat", fields, tags)
+	if strings.Contains(line, "=") {
+		kv := strings.Split(line, "=")
+		ival, err := strconv.ParseInt(kv[1], 10, 64)
+		if err == nil {
+			fields := map[string]interface{}{"total": ival}
+			acc.AddFields("redis_errorstat", fields, tags)
+		} else if log != nil {
+			log.Debugf("Error while parsing %v: %v", line, err)
+		}
+	} else if log != nil {
+		log.Debugf("Missing '=' in line %v (name:%v).", line, name)
```

Comment:
I really would prefer
```suggestion
	kv := strings.Split(line, "=")
	if len(kv) < 2 {
		acc.AddError(fmt.Errorf("invalid line for %q: %s", name, line))
		return
	}
	ival, err := strconv.ParseInt(kv[1], 10, 64)
	if err != nil {
		acc.AddError(fmt.Errorf("parsing value in line %q failed: %w", line, err))
		return
	}
	
	fields := map[string]interface{}{"total": ival}
	acc.AddFields("redis_errorstat", fields, tags)
```
and not passing the log.